src/transformers/modeling_utils.py

```diff
@@ -1154,6 +1155,10 @@ def __init__(self, config: PretrainedConfig, *inputs, **kwargs):
         self.name_or_path = config.name_or_path
         self.warnings_issued = {}
         self.generation_config = GenerationConfig.from_model_config(config) if self.can_generate() else None
+        # Overwrite the class attribute to make it an instance attribute, so models like
+        # `InstructBlipForConditionalGeneration` can dynamically update it without modifying the class attribute
+        # when a different component (e.g. language_model) is used.
+        self._keep_in_fp32_modules = copy.copy(self.__class__._keep_in_fp32_modules)
```

Comment:
We can't do `self._keep_in_fp32_modules = self.__class__._keep_in_fp32_modules `, otherwise the in place operation like `extend` will modify the same object.